README.md

```diff
@@ -27,165 +27,155 @@ curl http://my-url
-		fmt.Fprint(w, "Hello, World!\n")
-	}
-	```
-
-	> Note that you can use any file name or package name (convention is to make
-	package name same as directory name).
-
-1. Now go to the `cmd` subdirectory.
-	```sh
-	cd cmd
-	```
-
-1. Create a `main.go` file with the following contents:
-	```golang
-	package main
-
-	import (
-		"log"
-		"os"
-
-		"github.com/GoogleCloudPlatform/functions-framework-go/funcframework"
-		"example.com/hello"
-	)
-
-	func main() {
-		if err := funcframework.RegisterHTTPFunction("/", hello.HelloWorld); err != nil {
-			log.Fatalf("funcframework.RegisterHTTPFunction: %v\n", err)
-		}
-		// Use PORT environment variable, or default to 8080.
-		port := "8080"
-		if envPort := os.Getenv("PORT"); envPort != "" {
-			port = envPort
-		}
-
-		if err := funcframework.Start(port); err != nil {
-			log.Fatalf("funcframework.Start: %v\n", err)
-		}
-	}
-	```
-
-1. Start the local development server:
-	```sh
-	go build
-	./cmd
-	Serving function...
-	```
-
-2. Send requests to this function using `curl` from another terminal window:
-	```sh
-	curl localhost:8080
-	# Output: Hello, World!
-	```
-
-# Run your function on serverless platforms
-
-## Google Cloud Functions
-
-You cannot deploy main packages to Google Cloud Functions. You need to go back to the parent directory
-in which your function code is.
+## Quickstart: Hello, World on your local machine
+
+1.  Make sure you have Go 1.11+ installed with: `go version` The output should
+    be Go 1.11 or higher.
+
+2.  Create the necessary directories. `sh mkdir -p hello/cmd cd hello`
+
+3.  Create a Go module: `sh go mod init example.com/hello`
+
+    > Note: You can use a different module name rather than `example.com/hello`.
+
+4.  Create a `function.go` file with the following contents: ```golang package
+    hello
+
+    import ( "net/http" "fmt" )
+
+    // HelloWorld writes "Hello, World!" to the HTTP response. func HelloWorld(w
+    http.ResponseWriter, r *http.Request) { fmt.Fprint(w, "Hello, World!\n") }
+    ```
+
+    > Note that you can use any file name or package name (convention is to make
+    > package name same as directory name).
+
+1.  Now go to the `cmd` subdirectory. `sh cd cmd`
+
+2.  Create a `main.go` file with the following contents:
+
+```golang
+package main
+
+import (
+  "log"
+  "os"
+  "github.com/GoogleCloudPlatform/functions-framework-go/funcframework"
+  "example.com/hello"
+)
+
+func main() {
+  ctx := context.Backgroun()
+  if err := funcframework.RegisterHTTPFunctionContext("/", hello.HelloWorld); err != nil {
+    log.Fatalf("funcframework.RegisterHTTPFunction: %v\n", err)
+  }
+
+  // Use PORT environment variable, or default to 8080.
+  port := "8080"
+  if envPort := os.Getenv("PORT"); envPort != "" {
+    port = envPort
+  }
+
+  if err := funcframework.Start(port); err != nil {
+    log.Fatalf("funcframework.Start: %v\n", err)
+  }
+}
+```
+
+1.  Start the local development server: `sh go build ./cmd Serving function...`
```

Comment:
This command doesn't make sense. Was an autoformatter used?

Can we have the previous instructions that seemed fine?

Is there a reason for breaking into another shell with `sh`?